core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1075,7 +1079,8 @@ private static RelDataType arrayReturnType(SqlOperatorBinding opBinding) {
   public static final SqlFunction ARRAY =
       SqlBasicFunction.create("ARRAY",
           SqlLibraryOperators::arrayReturnType,
-          OperandTypes.SAME_VARIADIC);
+          OperandTypes.SAME_VARIADIC,
+          SqlFunctionCategory.SYSTEM);
```

Comment:
what is meaning of System function?And why array function belong to system function.

Reply:
current the function type of ARRAY is Numeric currently. Obviously not right. The closest type in calcite is System. it means built-in or system-in. we can see many other functions if not belongs to timedate/string/numeric will let it be System.